src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useContext } from 'react';
+import { Filter } from '../Filter/Filter';
+import { TodoContext } from '../Store/TodoContext';
+import { deleteTodos } from '../../api/todos';
+
+export const Footer: React.FC = () => {
+  const { todos, setTodos, setErrorMessage } = useContext(TodoContext);
+
+  const hasEnoughTodos = todos.length > 0;
+
+  const uncompletedTodos = todos.filter(todo => !todo.completed);
+
+  const hasEnoughCompletedTodo = todos.some(todo => todo.completed);
+
+  const todoInput = document.getElementById('todoInput');
+
+  const handleDelete = (todoId: number) => {
+    deleteTodos(todoId)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.filter(post => post.id !== todoId),
+        );
+      })
+      .catch(error => {
+        setErrorMessage('Unable to delete a todo');
+        setTodos(todos);
+        throw error;
+      })
+      .finally(() => {
+        if (todoInput) {
+          todoInput.focus();
+        }
+      });
+  };
+
+  const handleTodoCleaning = () => {
+    todos.map(todo => todo.completed && handleDelete(todo.id));
+    setTodos(todos.filter(todo => !todo.completed));
```

Comment:
```suggestion
    const completedTodos = todos.filter(todo => todo.completed);
    completedTodos.forEach(todo => handleDelete(todo.id));
```
looks like it is reason of fail this test.
you are already filter todos in then block.
![Screenshot from 2024-03-17 15-27-05](https://github.com/mate-academy/react_todo-app-add-and-delete/assets/33029864/2de3139f-2ba3-4d62-80fd-e69445a3acb9)

Reply:
The problem is that one day it passes and the next it doesn't.